From 8a9efba3d1374427866466e0aca623590bda64e3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:24 -0400
Subject: [usb] xhci: avoid warning for !PM_SLEEP

Message-id: <1410887463-20674-493-git-send-email-dzickus@redhat.com>
Patchwork-id: 95469
O-Subject: [RHEL7 PATCH 492/591] usb: xhci: avoid warning for !PM_SLEEP
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 274f6afa298791df97fd37b7bc9d8327f5cf6ee9
Author: Arnd Bergmann <arnd@arndb.de>
Date:   Thu May 8 15:52:19 2014 +0200

    usb: xhci: avoid warning for !PM_SLEEP

    If we build a kernel with PM_SUSPEND set and no PM_SLEEP,
    we get a build warning in the xhci-plat driver about unused
    functions.

    To fix this, use "#ifdef CONFIG_PM_SLEEP", like we do in most
    other drivers nowadays.

    Signed-off-by: Arnd Bergmann <arnd@arndb.de>
    Cc: Mathias Nyman <mathias.nyman@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 287359c..3fce2c7 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -223,7 +223,7 @@ static int xhci_plat_remove(struct platform_device *dev)
  return 0;
 }
 
-#ifdef CONFIG_PM
+#ifdef CONFIG_PM_SLEEP
 static int xhci_plat_suspend(struct device *dev)
 {
  struct usb_hcd *hcd = dev_get_drvdata(dev);
-- 
1.7.1